### PR TITLE
Delombok plugin API and add JSR-305 annotations

### PR DESCRIPTION
**Pull request by dmarcuse**

### CODE REVIEW

File: plugin-api/src/main/java/net/clgd/ccemux/api/config/Config.java [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+	private final Group root = new Group();
+
 	/**
 	 * The root group for the config. All properties and child groups are stored in
 	 * this.
 	 */
-	@Getter
-	private final Group root = new Group();
+	@Nonnull
+	public Group getRoot() {
```
Comment (SquidDev) [inline plugin-api/src/main/java/net/clgd/ccemux/api/config/Config.java:18]: There's a bit of me which would just call this `root()`, though I realise it doesn't fit java beans or w/e.

File: plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedPalette.java [hunk 2]
```diff
@@ -90,21 +78,38 @@ public void resetColours() {
...
+	 */
+	@SuppressWarnings("all")
+	public boolean isChanged() {
+		return this.changed;
+	}
+
+	/**
+	 * Whether a color in this palette has been changed
+	 */
+	@SuppressWarnings("all")
```
Comment (SquidDev) [inline plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedPalette.java]: Why is this needed? I assume we're doing something odd with CC interaction?

File: plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualFile.java [hunk 3]
```diff
@@ -2,34 +2,60 @@
...
 	}
+
+	@Override
+	@SuppressWarnings("all")
+	public boolean equals(final java.lang.Object o) {
+		if (o == this) return true;
+		if (!(o instanceof VirtualFile)) return false;
+		final VirtualFile other = (VirtualFile) o;
+		if (!other.canEqual((java.lang.Object) this)) return false;
+		if (!java.util.Arrays.equals(this.getData(), other.getData())) return false;
```
Comment (SquidDev) [inline plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualFile.java]: Can we remove the qualified usage?

File: plugin-api/src/main/java/net/clgd/ccemux/api/plugins/Plugin.java [hunk 4]
```diff
@@ -22,8 +22,8 @@
  * @author apemanzilla
  * @see Hook
  */
-@Slf4j
 public abstract class Plugin {
+	private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(Plugin.class);
```
Comment (SquidDev) [inline plugin-api/src/main/java/net/clgd/ccemux/api/plugins/Plugin.java]: Again, import instead of qualified usage.

**Review by SquidDev — CHANGES_REQUESTED** [review]

File: plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedPalette.java (hunk 2 above)
Comment (dmarcuse) [reply plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedPalette.java]: It was generated by lombok and I missed it

File: plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualDirectory.java [hunk 5]
```diff
@@ -91,15 +86,45 @@ public VirtualDirectory() {
...
 	}
+
+	@Override
+	@SuppressWarnings("all")
+	public boolean equals(final java.lang.Object o) {
+		if (o == this) return true;
+		if (!(o instanceof VirtualDirectory)) return false;
+		final VirtualDirectory other = (VirtualDirectory) o;
+		if (!other.canEqual((java.lang.Object) this)) return false;
+		final java.lang.Object this$children = this.children;
```
Comment (SquidDev) [inline plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualDirectory.java]: `lang.Object`. Also `this$children` feels … blughr.

File: plugin-api/src/main/java/net/clgd/ccemux/api/rendering/PaletteAdapter.java [hunk 6]
```diff
@@ -77,4 +76,49 @@ public C getColor(int c) {
...
+	public ColorAdapter<C> getAdapter() {
+		return this.adapter;
+	}
+
+	@java.lang.Override
+	public boolean equals(@Nonnull Object o) {
+		if (o == this) return true;
+		if (!(o instanceof PaletteAdapter)) return false;
+		final PaletteAdapter<?> other = (PaletteAdapter<?>) o;
+		final java.lang.Object this$palette = this.getPalette();
```
Comment (SquidDev) [inline plugin-api/src/main/java/net/clgd/ccemux/api/rendering/PaletteAdapter.java]: Same here.

**Review by SquidDev — CHANGES_REQUESTED** [review]
Sorry, a couple more nit-picks.

File: plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualDirectory.java (hunk 5 above)
Comment (dmarcuse) [reply plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualDirectory.java]: I'm tempted to just use Kotlin and data classes to avoid this ugliness....

Comment (SquidDev) [reply plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualDirectory.java]: There's a reason your IDE has code-generators for this! It's write-only code.

**Review by SquidDev — COMMENTED** [review]
LGTM

**Review by SquidDev — APPROVED** [review]

**Comment by SquidDev** [comment]
I lied. There's things I need to fix :p.
